lib/expect.toit

```diff
@@ -117,11 +117,12 @@ expect_bytes_equal expected/ByteArray actual/ByteArray:
 /** Expects $throw_block to throw an object equal to the $expected value. */
 expect_throw expected [throw_block]:
   try:
-     throw_block.call
-  finally: | is_exception exception |
-    if not is_exception:
+    error := catch --trace=(: expected != it) --unwind=(: expected != it):
+      throw_block.call
+    if expected != error:
```

Comment:
This doesn't make sense to me.

If the caught exception is not equal to the `expected` we have a message that says that we got `null`?

I think this test was correctly testing whether `error` was null`. That said: given that an exception can be a boolean, making it an explicit `error == null` is clearly better.

Also: I wouldn't use the `--message` thingy anymore. Just `throw "Expected exception, but executed normally"` or so.